--- /home/andre/.local/pipx/venvs/ansible/lib/python3.10/site-packages/ansible/../ansible_collections/community/general/plugins/modules/cloud/lxd/lxd_container.py	2022-07-15 08:27:36.697448545 +0200
+++ ../../library/lxd_container.py	2022-07-15 08:32:52.460497230 +0200
@@ -131,6 +131,15 @@
         required: false
         default: false
         type: bool
+    wait_for_ipv4_interfaces:
+        description:
+          - A list of interfaces for which C(wait_for_ipv4_addresses) will
+            wait.  This is useful if the container is running a nested docker,
+            in which case it might have interfaces which never acquires ipv4
+            addresses.
+        required: false
+        default: [] (wait for all interfaces)
+        type: list
     wait_for_container:
         description:
             - If set to C(true), the tasks will wait till the task reports a
@@ -446,6 +455,7 @@
 
         self.timeout = self.module.params['timeout']
         self.wait_for_ipv4_addresses = self.module.params['wait_for_ipv4_addresses']
+        self.wait_for_ipv4_interfaces = self.module.params['wait_for_ipv4_interfaces']
         self.force_stop = self.module.params['force_stop']
         self.addresses = None
         self.target = self.module.params['target']
@@ -569,6 +579,8 @@
         resp_json = self._get_instance_state_json()
         network = resp_json['metadata']['network'] or {}
         network = dict((k, v) for k, v in network.items() if k not in ignore_devices) or {}
+        if self.wait_for_ipv4_interfaces:
+            network = dict((k, v) for k, v in network.items() if k in self.wait_for_ipv4_interfaces)
         addresses = dict((k, [a['address'] for a in v['addresses'] if a['family'] == 'inet']) for k, v in network.items()) or {}
         return addresses
 
@@ -806,6 +818,9 @@
                 type='bool',
                 default=False
             ),
+            wait_for_ipv4_interfaces=dict(
+                type='list',
+            ),
             force_stop=dict(
                 type='bool',
                 default=False
@@ -831,16 +846,6 @@
         supports_check_mode=False,
     )
 
-    if module.params['ignore_volatile_options'] is None:
-        module.params['ignore_volatile_options'] = True
-        module.deprecate(
-            'If the keyword "volatile" is used in a playbook in the config'
-            'section, a "changed" message will appear with every run, even without a change'
-            'to the playbook.'
-            'This will change in the future. Please test your scripts'
-            'by "ignore_volatile_options: false". To keep the old behavior, set that option explicitly to "true"',
-            version='6.0.0', collection_name='community.general')
-
     lxd_manage = LXDContainerManagement(module=module)
     lxd_manage.run()
 
