--- /home/andre/.local/pipx/venvs/ansible/lib/python3.10/site-packages/ansible/../ansible_collections/community/general/plugins/modules/lxd_container.py	2022-11-09 06:39:00.096258530 +0100
+++ ../../library/lxd_container.py	2022-11-22 07:42:30.303333877 +0100
@@ -132,6 +132,15 @@
         required: false
         default: false
         type: bool
+    wait_for_ipv4_interfaces:
+        description:
+          - A list of interfaces for which C(wait_for_ipv4_addresses) will
+            wait.  This is useful if the container is running a nested docker,
+            in which case it might have interfaces which never acquires ipv4
+            addresses.
+        required: false
+        default: [] (wait for all interfaces)
+        type: list
     wait_for_container:
         description:
             - If set to C(true), the tasks will wait till the task reports a
@@ -447,6 +456,7 @@
 
         self.timeout = self.module.params['timeout']
         self.wait_for_ipv4_addresses = self.module.params['wait_for_ipv4_addresses']
+        self.wait_for_ipv4_interfaces = self.module.params['wait_for_ipv4_interfaces']
         self.force_stop = self.module.params['force_stop']
         self.addresses = None
         self.target = self.module.params['target']
@@ -570,6 +580,8 @@
         resp_json = self._get_instance_state_json()
         network = resp_json['metadata']['network'] or {}
         network = dict((k, v) for k, v in network.items() if k not in ignore_devices) or {}
+        if self.wait_for_ipv4_interfaces:
+            network = dict((k, v) for k, v in network.items() if k in self.wait_for_ipv4_interfaces)
         addresses = dict((k, [a['address'] for a in v['addresses'] if a['family'] == 'inet']) for k, v in network.items()) or {}
         return addresses
 
@@ -808,6 +820,9 @@
                 type='bool',
                 default=False
             ),
+            wait_for_ipv4_interfaces=dict(
+                type='list',
+            ),
             force_stop=dict(
                 type='bool',
                 default=False
