diff --git a/src/mailman/app/moderator.py b/src/mailman/app/moderator.py
index aa044b83d456d9ac02fbab05d93c148ce36f8d67..898b14160d0039a37b3e2cf26856c6f4ce3c4561 100644
--- a/src/mailman/app/moderator.py
+++ b/src/mailman/app/moderator.py
@@ -204,8 +204,9 @@ def handle_message(mlist, id, action, comment=None, forward=None):
         # request for it.
         delete = True
         for token, data in pendings.find(pend_type='data'):
-            if data['_mod_message_id'] == message_id:
+            if data and data.get('_mod_message_id') == message_id:
                 delete = False
+                break
         if delete:
             message_store.delete_message(message_id)
     # Log the rejection
